### PR TITLE
5082 fix patch tx rule

### LINKED ISSUE #5082: [Bug][Azure DevOps] Error when editing the deployment regex of an existing transformation rule
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

**Version v0.17.0-beta3**

When I try to edit the deployment regex of an existing Azure DevOps transformation rule, Config-UI errors out and jumps to the page in the 2nd screenshot below. The backend error is shown in the 3rd screenshot.

Screen Shot 2023-05-03 at 9 46 29 PM
Screen Shot 2023-05-03 at 6 55 02 PM
Screen Shot 2023-05-03 at 6 55 23 PM

### What do you expect to happen

See above

### How to reproduce

See above

### Anything else

_No response_

### Version

main

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by CamilleTeruel**
### Summary

Fix the PATCH endpoint of remote plugin transformation rules API.
Refactor and add tests

### Does this close any open issues?
Closes #5082

### CODE REVIEW

File: backend/test/helper/api.go [hunk 1]
```diff
@@ -134,43 +136,43 @@ func (d *DevlakeClient) ListProjects() apiProject.PaginatedProjects {
 	}, http.MethodGet, fmt.Sprintf("%s/projects", d.Endpoint), nil, nil)
 }
 
-func (d *DevlakeClient) CreateScope(pluginName string, connectionId uint64, scopes ...any) any {
+func (d *DevlakeClient) CreateScope(pluginName string, connectionId uint64, scopes ...any) []Map {
```
Comment (keon94) [inline backend/test/helper/api.go:140]: Semantically, these generic client APIs are meant to be used in conjunction with the helper.Cast[Type] method, e.g.
```helper.Cast[[]AzureDevopsGitRepo](client.CreateScope(azurePlugin, connection.ID, scopes))```

Comment (keon94) [reply backend/test/helper/api.go:140]: Here's how it's looking in my PR (#4906):
https://github.com/merico-dev/lake/blob/ce59afdf3fbec73060c16f62e062362e737c8371/backend/test/e2e/remote/helper.go#L82

Comment (CamilleTeruel) [reply backend/test/helper/api.go]: Alright, I removed the test refactor commit and used this helper function instead.

**Review by hezyin — APPROVED** [review]
LGTM

**Comment by hezyin** [comment]
This fix needs to be cherry-picked back to release-v0.17

**Pull request merged**